lib/sdr_client/redesigned_client/request_builder.rb

```diff
@@ -0,0 +1,171 @@
+# frozen_string_literal: true
+
+module SdrClient
+  class RedesignedClient
+    # Builds a Cocina request object from metadata. This is what we send to the server when doing a deposit.
+    class RequestBuilder # rubocop:disable Metrics/ClassLength
+      # @param [String] apo the object ID of the administrative policy object
+      # @param [String] source_id the source ID of the object
+      # @param [Hash] options optional parameters
+      # @option options [String] label the required object label
+      # @option options [String] view the access level for viewing the object
+      # @option options [String] download the access level for downloading the object
+      # @option options [String] type (https://cocina.sul.stanford.edu/models/object) the required object type.
+      # @option options [String] use_and_reproduction the use and reproduction statement
+      # @option options [String] copyright the copyright statement
+      # @option options [String] collection the object ID of the collection object
+      # @option options [String] catkey the catalog key (from now unused Symphony ILS)
+      # @option options [String] folio_instance_hrid the instance ID from the Folio ILS
+      # @option options [String] viewing_direction the viewing direction (left to right, right to left)
+      # @option options [Date|nil] embargo_release_date when to release the embargo (or nil if none)
+      # @option options [String] embargo_access the access level for viewing the object after the embargo period
+      # @option options [String] embargo_download the access level for downloading the object after the embargo period
+      # @option options [Array<FileSet>] file_sets the file sets to attach.
+      # @option options [Hash<String, Hash<String, String>>] files_metadata file name, hash of additional file metadata
+      # Additional metadata includes access, preserve, shelve, publish, md5, sha1
+      def initialize(apo:, source_id:, **options)
+        @apo = apo
+        @source_id = source_id
+        @options = options
```

Comment:
Move all non-required options to an options param instead of having a giant list of keyword params.